### PR TITLE
cozy-drive: Add Cozy Drive package.

### PR DESCRIPTION
**Pull request by Simarra**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -12572,5 +12572,11 @@
     github = "mbprtpmnr";
     githubId = 88109321;
   };
+  simarra = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:12575]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 2]
```diff
@@ -264,6 +264,8 @@ subsonic-compatible api. Available as [navidrome](#opt-services.navidrome.enable
 
 - The `isabelle` package has been upgraded from 2020 to 2021
 
+- The `cozy-drive` package has been added. v3.29.0.
+
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2111.section.md:268]:
```suggestion
```

I don't think this change is significant to over 99% of the users and so does not require a changelog entry. Also we usually only announce new package if they are really important to a big group of users.

File: pkgs/applications/networking/cozy-drive/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+in
+appimageTools.wrapType2 {
+  inherit name src;
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cozy-drive/default.nix:25]: I wouldn't mind if we used less pname here.

File: pkgs/applications/networking/cozy-drive/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  '';
+
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cozy-drive/default.nix:29]:
```suggestion
  '';

  meta = with lib; {
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the contributing guide when naming your commits.

**Comment by Simarra** [comment]
Everything should be ok now.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml [hunk 5]
```diff
@@ -847,6 +847,12 @@ Superuser created successfully.
           2020 to 2021
         </para>
       </listitem>
+      <listitem>
+        <para>
+          The <literal>cozy-drive</literal> package has been added.
+          v3.29.0.
+        </para>
+      </listitem>
```
Comment (bobby285271) [inline nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:855]: Please drop this, otherwise the NixOS manual checks will fail.
